examples/optimistic-ui/pages/index.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import fetch from '../libs/fetch'
+
+import useSWR, { mutate } from '@zeit/swr'
+
+export default () => {
+  const [text, setText] = React.useState('');
+  const { data } = useSWR('/api/data', fetch)
+
+  async function handleSubmit(event) {
+    event.preventDefault()
+    // mutate current data to optimistically update the UI
+    // the fetch below could fail, in that case the UI will
+    // return to the previous state after SWR revalidate it
+    mutate('/api/data', [...data, text])
+    // send text to the API
+    await fetch('/api/data', {
+      method: 'POST',
+      body: JSON.stringify({ text })
+    })
```

Comment:
If we mutate & revalidate before sending the request, the revalidation request (GET) might arrive earlier than the API request (POST), so it could still be out to dated.

Instead, I think we can do this if we want to do local mutation optimistically:
```js
import { mutate, trigger } from '@zeit/swr'

mutate('/api/data', data, false) // set `shouldRevalidate` to false
await fetch(API)
trigger('/api/data') // trigger a revalidation manually
```